src/umpire/util/FixedMallocPool.hpp

```diff
@@ -42,6 +53,7 @@ class FixedMallocPool {
 
   void newPool();
   void* allocInPool(Pool& p) noexcept;
+  void* allocate_impl(std::size_t bytes);
```

Comment:
What is the purpose for `allocate_impl` instead of `allocate`?  It seems different from the rest of the patterns we use